arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -89,6 +153,22 @@ final class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdate
      */
     var initialTransformationMatrix: TransformationMatrix = TransformationMatrix.createIdentityMatrix()
 
+    /**
+     * A quaternion used to compensate for the pitch being 90 degrees on ARCore; used to calculate the current device
+     * transformation for each frame.
+     *
+     * @since 100.6.0
+     */
+    private val compensationQuaternion: Quaternion =
+        Quaternion((sin(45 / (180 / PI)).toFloat()), 0F, 0F, (cos(45 / (180 / PI)).toFloat()))
+
+    /**
+     * Initial location from [locationDataSource].
+     *
+     * @since 100.6.0
+     */
+    private var initialLocation: Point? = null
```

Comment:
But as far as I can tell, you just assign a value or null to `initialLocation`, but never actually use this property anywhere. Seems redundant to me, maybe I'm missing something?

Reply:
We use it to check if we've ever received an `initialLocation`. If we haven't, we set a new origin camera onto the camera controller. If we have and we aren't using AR core, we just update the camera's location. This will become more clear when we implement the changes to check for distance between the initial location and the current location to reset the initial location but it's still relevant here.

Reply:
Ok, sounds ok. 
The part I don't understand is why we don't also use the `cameraController` when we don't use ARCore. I don't see any code that clears the `cameraController` if we don't use ARCore. We only seem to set a `cameraController` in `initialize()` and then never touch it again.

Reply:
The whole "not using ARCore" scenario is not part of min ship for U100.6 and won't be implemented until the next release.  The code above is the same as on iOS and part of the original implementation when we thought it would be part of this release.